src/contexts/ServerContext.js

```diff
@@ -43,34 +43,136 @@ export const ServerProvider = ({ children }) => {
     const [specificQueryContext, setSpecificQueryContext] = useState(initialState)
     const [specificQueryTotal, setSpecificQueryTotal] = useState(0)
 
-    const loadUsers = (offset, limit, name, order, orderBy) => {
+    const loadUsers = async (offset, limit, order, orderBy, name) => {
         setIsLoadingUserContext(true);
-        let _query = `SELECT USER_NAME, USER_TYPE, USER_ZONE WHERE USER_TYPE != 'RODSGROUP'`;
-        if (name !== '') {
-            _query = `SELECT USER_NAME, USER_TYPE, USER_ZONE WHERE USER_TYPE != 'RODSGROUP' and USER_NAME LIKE '%${name.toUpperCase()}%'`
-        }
-        _query = queryGenerator(_query, order, orderBy);
-        return axios({
-            method: 'GET',
-            url: `${restApiLocation}/query`,
-            headers: {
-                'Authorization': localStorage.getItem('zmt-token')
-            },
-            params: {
-                query: _query,
-                limit: limit,
-                offset: offset,
-                type: 'general',
-                'case-sensitive': 0
+
+    
+        if (name == "" || name == "USER_NAME") {
+            let _query = `SELECT USER_NAME, USER_TYPE, USER_ZONE WHERE USER_TYPE != 'RODSGROUP'`;
+
+            _query = queryGenerator(_query, order, orderBy);
+
+            return axios({
+                method: 'GET',
+                url: `${restApiLocation}/query`,
+                headers: {
+                    'Authorization': localStorage.getItem('zmt-token')
+                },
+                params: {
+                    query: _query,
+                    limit: limit,
+                    offset: offset,
+                    type: 'general',
+                    'case-sensitive': 0
+                }
+            }).then((res) => {
+                if (name === '') setUserTotal(res.data.total)
+                setUserTotal(res.data.total)
+                setUserContext(res.data);
+                setIsLoadingUserContext(false);
+            }).catch(() => {
+                setUserContext(undefined);
+                setIsLoadingUserContext(false);
+            });
+        } else {
+            let totalData = [];
+
+            let queryTry1 = `SELECT USER_NAME, USER_TYPE, USER_ZONE WHERE USER_TYPE != 'RODSGROUP' AND USER_NAME LIKE '%${name.toUpperCase()}%'`
+            let q1 = queryGenerator(queryTry1, order, orderBy);
+
+            let queryTry2 = `SELECT USER_NAME, USER_TYPE, USER_ZONE WHERE USER_TYPE != 'RODSGROUP' AND USER_ZONE LIKE '%${name.toUpperCase()}%'`
+            let q2 = queryGenerator(queryTry2, order, orderBy);
+
+            let queryTry3 = `SELECT USER_NAME, USER_TYPE, USER_ZONE WHERE USER_TYPE != 'RODSGROUP' AND USER_TYPE LIKE '%${name.toUpperCase()}%'`
+            let q3 = queryGenerator(queryTry3, order, orderBy);
```

Comment:
these queries look identical...

why do we run them three times?   i don't remember.

Reply:
oh!  i spy.  nevermind.

Reply:
Also, does `RODSGROUP` actually work?

Reply:
Is the reason for the all caps rodsgroup related to case-insensitivity?

Reply:
Yep, `RODSGROUP` actually works. If we just had `SELECT USER_NAME, USER_TYPE, USER_ZONE`, we get back `["testGroup","rodsgroup","tempZone"]` included in the response. Additionally, `RODSGROUP` being capitalized is just for consistency with the `toUpperCase()` later in the statement. Also, we need to use case-insensitive because if someone filtered for `uSeR9` and `user9` was in the system, it wouldn't be returned if it was case sensitive.

Though I suppose that since we're already using case sensitive, we can get rid of the toUpperCase() and make rodsgroup lower case, with no effect on the results. Is that something we want to do?

Reply:
Going to remove capitalization, can revert if needed.

Reply:
i think the search itself can be case-insensitive... but the results should definitely be shown and selection made by the administrator in the GUI, based on the case-sensitive version... as they are distinct users/groups/zones.

Reply:
If I understand this correctly, you're saying that searching for `usER9` and `user9` should return the same list of users back (case insensitive search), but on the frontend, we need to still keep a user like `daVeBoB` displayed with its capitalization?

If so, that's what we have right now

Reply:
Yes, that is what I was saying.  Great.